pallets/drand/src/lib.rs

```diff
@@ -599,10 +599,21 @@ impl<T: Config> Pallet<T> {
 	/// Query the endpoint `{api}/{chainHash}/info` to receive information about the drand chain
 	/// Valid response bodies are deserialized into `BeaconInfoResponse`
 	fn fetch_drand_chain_info() -> Result<String, http::Error> {
+		let uri: &str = &format!("{}/{}/info", API_ENDPOINT, QUICKNET_CHAIN_HASH);
+		Self::fetch(uri)
+	}
+
+	/// fetches the latest randomness from drand's API
+	fn fetch_drand() -> Result<String, http::Error> {
+		let uri: &str = &format!("{}/{}/public/latest", API_ENDPOINT, QUICKNET_CHAIN_HASH);
+		Self::fetch(uri)
+	}
+
+	/// Fetch a remote URL and return the body of the response as a string.
+	fn fetch(uri: &str) -> Result<String, http::Error> {
 		// TODO: move this value to config
 		// https://github.com/ideal-lab5/pallet-drand/issues/5
-		let deadline = sp_io::offchain::timestamp().add(Duration::from_millis(1_000));
-		let uri: &str = &format!("{}/{}/info", API_ENDPOINT, QUICKNET_CHAIN_HASH);
+		let deadline = sp_io::offchain::timestamp().add(Duration::from_millis(2_000));
```

Comment:
What if we made the wait time configurable? something like T::HttpFetchTimeout?

Reply:
yes, that's this issue https://github.com/ideal-lab5/pallet-drand/issues/5
but good point, we can include it as part of it